File: Test/AdventureWorksFunctionalModel/Mapping/VendorMap.cs
using System.Data.Entity.ModelConfiguration;

using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace AW.Mapping {
    public class VendorMap : EntityTypeConfiguration<Vendor> {
        public VendorMap() {
            // Primary Key
            HasKey(t => t.BusinessEntityID);

            // Properties
            Property(t => t.AccountNumber)
                .IsRequired()
                .HasMaxLength(15);

            Property(t => t.Name)
                .IsRequired()
                .HasMaxLength(50);

            Property(t => t.PurchasingWebServiceURL)
                .HasMaxLength(1024);

            // Table & Column Mappings
            ToTable("Vendor", "Purchasing");
            Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            Property(t => t.AccountNumber).HasColumnName("AccountNumber");
            Property(t => t.Name).HasColumnName("Name");
            Property(t => t.CreditRating).HasColumnName("CreditRating");
            Property(t => t.PreferredVendorStatus).HasColumnName("PreferredVendorStatus");
            Property(t => t.ActiveFlag).HasColumnName("ActiveFlag");
            Property(t => t.PurchasingWebServiceURL).HasColumnName("PurchasingWebServiceURL");
            Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }

    public static partial class Mapper {
        public static void Map(this EntityTypeBuilder<Vendor> builder) {
            builder.HasKey(t => t.BusinessEntityID);

            // Properties
            builder.Property(t => t.AccountNumber)
                   .IsRequired()
                   .HasMaxLength(15);

            builder.Property(t => t.Name)
                   .IsRequired()
                   .HasMaxLength(50);

            builder.Property(t => t.PurchasingWebServiceURL)
                   .HasMaxLength(1024);

            // Table & Column Mappings
            builder.ToTable("Vendor", "Purchasing");
            builder.Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            builder.Property(t => t.AccountNumber).HasColumnName("AccountNumber");
            builder.Property(t => t.Name).HasColumnName("Name");
            builder.Property(t => t.CreditRating).HasColumnName("CreditRating");
            builder.Property(t => t.PreferredVendorStatus).HasColumnName("PreferredVendorStatus");
            builder.Property(t => t.ActiveFlag).HasColumnName("ActiveFlag");
            builder.Property(t => t.PurchasingWebServiceURL).HasColumnName("PurchasingWebServiceURL");
            builder.Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }
}

File: Test/AdventureWorksLegacyModelVB/Mapping/VendorMap.vb
Imports System.Data.Entity.ModelConfiguration

Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders

Namespace AW.Mapping
	Public Class VendorMap
		Inherits EntityTypeConfiguration(Of Vendor)

		Public Sub New()
			' Primary Key
			HasKey(Function(t) t.BusinessEntityID)

			' Properties
			[Property](Function(t) t.AccountNumber).IsRequired().HasMaxLength(15)

			[Property](Function(t) t.Name).IsRequired().HasMaxLength(50)

			[Property](Function(t) t.PurchasingWebServiceURL).HasMaxLength(1024)

			' Table & Column Mappings
			ToTable("Vendor", "Purchasing")
			[Property](Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			[Property](Function(t) t.AccountNumber).HasColumnName("AccountNumber")
			[Property](Function(t) t.Name).HasColumnName("Name")
			[Property](Function(t) t.CreditRating).HasColumnName("CreditRating")
			[Property](Function(t) t.PreferredVendorStatus).HasColumnName("PreferredVendorStatus")
			[Property](Function(t) t.ActiveFlag).HasColumnName("ActiveFlag")
			[Property](Function(t) t.PurchasingWebServiceURL).HasColumnName("PurchasingWebServiceURL")
			[Property](Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Class

	Public Module Mapper
		<System.Runtime.CompilerServices.Extension> _
		Public Sub Map(ByVal builder As EntityTypeBuilder(Of Vendor))
			builder.HasKey(Function(t) t.BusinessEntityID)

			' Properties
			builder.Property(Function(t) t.AccountNumber).IsRequired().HasMaxLength(15)

			builder.Property(Function(t) t.Name).IsRequired().HasMaxLength(50)

			builder.Property(Function(t) t.PurchasingWebServiceURL).HasMaxLength(1024)

			' Table & Column Mappings
			builder.ToTable("Vendor", "Purchasing")
			builder.Property(Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			builder.Property(Function(t) t.AccountNumber).HasColumnName("AccountNumber")
			builder.Property(Function(t) t.Name).HasColumnName("Name")
			builder.Property(Function(t) t.CreditRating).HasColumnName("CreditRating")
			builder.Property(Function(t) t.PreferredVendorStatus).HasColumnName("PreferredVendorStatus")
			builder.Property(Function(t) t.ActiveFlag).HasColumnName("ActiveFlag")
			builder.Property(Function(t) t.PurchasingWebServiceURL).HasColumnName("PurchasingWebServiceURL")
			builder.Property(Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Module
End Namespace